### ISSUE TITLE
illegal hardware instruction

### ISSUE THREAD

**Issue by teepeemm** [vectorgraphics/asymptote#217]
I've encountered a strange situation.  If I run `asy filename.asy` on the following:
```
settings.tex = "xelatex";
import graph3;
usepackage("mathspec");
texpreamble("\setallmainfonts{Calibri}");
real[] mylabels={60};
xaxis3("",59,62,black,InTicks(mylabels));
```
then I get the error message `zsh: illegal hardware instruction  asy filename.asy` and my file is not created.  This happens with InTicks, OutTicks, or InOutTicks, but not NoTicks3, and not in 2 dimensions, and only if I label 60 (this last condition makes it not too hard to avoid the error).  If I run `asy -v -v -k filename.asy`, then the last two commands before the error are
```
gs -q -dNOCACHE -dNOPAUSE -dBATCH -dSAFER -sDEVICE=eps2write -sOutputFile=filename__.ps%d filename__.pdf
gs -q -dBATCH -P -dSAFER -sDEVICE=eps2write -sOutputFile=/dev/null filename_.ps
```
If I run that last command, I get some postscript type output (I assume).  I've attached the files from the `-k` run.
[asymptote.zip](https://github.com/vectorgraphics/asymptote/files/6118154/asymptote.zip)

My hardware is a MacBook Pro with a quad-core Intel Core i7.
`asy --version` 2.65
`gs --version` 9.50

**Comment by johncbowman** [comment]
Version 2.65 is obsolete and no longer supported. Please upgrade to the latest version, 2.69.
You also should upgrade to gs 9.52 or later (the current version is 9.53.3).

**Comment by johncbowman** [comment]
I am reopening this issue as even under the latest git version, your code generates a stack overflow:
gs -q -dBATCH -P -dSAFER -sDEVICE=ps2write -sOutputFile=/dev/null test_.ps
          if(intersections(g,z-delta--w+delta).length != 2) continue;
          ^
/u/bowman/asymptote/base/three_surface.asy: 556.11: runtime: Stack overflow
Abort (core dumped)

**Comment by johncbowman** [comment]
Fixed in 38fb7ff28e7e6a6ccc28770bb757ee6fecd66d5a. This will be in the upcoming 2.70 release in the next few days; in the meantime you should be able to use version 2.69 with an up-to-date copy of 
https://raw.githubusercontent.com/vectorgraphics/asymptote/HEAD/base/three_surface.asy

**Comment by johncbowman** [comment]
A better fix is in 3cac27568d17d48c71250fa8aad4e2f41aae3707.
